layout/Query.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const _ = require('lodash');
+
+class ServiceConfiguration {
+  constructor() {
+    this._score = [];
+    this._filter = [];
+  }
+
+  filter(view) {
+    this._filter.push(view);
+  }
+
+  score(view) {
+    this._score.push(view);
+  }
+
+  render(vs) {
```

Comment:
I hesitate to do that since that stops us from being about to call `super.render()` from within the subclass without an error being thrown.